frontend/apps/pablo/updaters/oracle/oracle.ts

```diff
@@ -5,16 +5,48 @@ import {
   setOraclePrice,
 } from "@/store/oracle/slice";
 import useStore from "@/store/useStore";
-import axios from "axios";
+import axios, { AxiosResponse } from "axios";
 import BigNumber from "bignumber.js";
 
+function coinGeckoApiUrl(): string {
+  switch (process.env.NODE_ENV) {
+    case "production":
+      return "https://pro-api.coingecko.com/api/v3/simple/price";
+    default:
+      return "https://api.coingecko.com/api/v3/simple/price";
+  }
+}
+
+function coinGeckoHeaders():
+  | { headers: { x_cg_pro_api_key: string } }
+  | undefined {
+  switch (process.env.NODE_ENV) {
+    case "production":
+      return { headers: { x_cg_pro_api_key: process.env.COINGECKO_KEY! } };
+    default:
+      return undefined;
+  }
+}
+
+function coingeckoRequest(
+  targetTokens: string[],
+  fiatCurrencies: string[]
+): Promise<AxiosResponse<any, any>> {
+  return axios.get(
+    `${coinGeckoApiUrl()}?ids=${targetTokens.join(
+      ","
+    )}&vs_currencies=${fiatCurrencies.join(",")}`,
+    coinGeckoHeaders()
+  );
+}
+
 export const subscription = useStore.subscribe(
   (state) => state.substrateTokens,
   (state) => {
     const { hasFetchedTokens, tokens } = state;
 
     if (hasFetchedTokens) {
-      let vs_currencies = oracleCurrencies;
+      let vs_currencies = [...oracleCurrencies];
```

Comment:
We can directly pass in multiple requirements for the subscribe to work: 

```tsx
export const subscription = useStore.subscribe(
  (state) => ({
    hasFetchedTokens: state.substrateTokens.hasFetchedTokens,
    tokens: state.substrateTokens.tokens,
  }),
  ({ hasFetchedTokens, tokens }) => {

``` 

Also if you have a hard requirement, we can trick it to pass in an equality function: 

```tsx
useStore.subscribe(
  (state) => ({
    a: true, // needToBeTrue
    b: 0
  }),
  ({a, b}) => {},
  { 
    equalityFn: (a, b) => a.a === true && a.b === b.b
  }
)

```

This is to remove the gated if statement at the beginning : 

```
~~if (hasFetchedTokens)~~